File: app/src/main/java/me/harshithgoka/youtubedl/YoutubeDL/Utils/Arg.java
package me.harshithgoka.youtubedl.YoutubeDL.Utils;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by harshithg on 17/1/18.
 */



public class Arg extends JSONObject {
    public static final String VAL = "val";
    public Arg() {
        super();
    }

    public Arg(String s) {
        super();
        try {
            this.put(VAL, s);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
    
    public Arg(int i) {
        super();
        try {
            this.put(VAL, i);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public Arg(JSONArray a) {
        super();
        try {
            this.put(VAL, a);
        }
        catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/me/harshithgoka/youtubedl/YoutubeDL/Utils/Arg.kt
package me.harshithgoka.youtubedl.YoutubeDL.Utils

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

/**
 * Created by harshithg on 17/1/18.
 */


class Arg : JSONObject {
    constructor() : super()

    constructor(s: String) : super() {
        try {
            this.put(VAL, s)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }

    constructor(i: Int) : super() {
        try {
            this.put(VAL, i)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }

    constructor(a: JSONArray) : super() {
        try {
            this.put(VAL, a)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }

    companion object {
        const val VAL = "val"
    }
}